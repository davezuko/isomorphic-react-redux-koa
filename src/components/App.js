import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import PropTypes from 'prop-types'

class App extends React.Component {
  static propTypes = {
    store: PropTypes.object.isRequired
  }

  shouldComponentUpdate () {
    return false
  }

  render () {
    return (
      <Provider store={this.props.store}>
        <div style={{ height: '100%' }}>
          <Router>
            <this.props.routes />
          </Router>
        </div>
      </Provider>
    )
  }
}

export default App
