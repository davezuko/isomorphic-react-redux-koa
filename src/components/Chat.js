import React, { PropTypes, Component } from 'react'

export default class Chat extends Component {
  static propTypes = {
    messages: PropTypes.array,
    sendMessage: PropTypes.func.isRequired,
    user: PropTypes.string.isRequired
  }

  constructor() {
    super()
    this.handleSendClick = this.handleSendClick.bind(this)
  }

  /**
   * Отправляет сообщение в чат
   */
  handleSendClick() {
    const inputText = this.refs.inputText.value.trim()

    if (inputText) {
      this.refs.inputText.value = ''

      console.log(this.props)
      this.props.sendMessage(this.props.user, inputText)
    }
  }

  render() {
    const { messages } = this.props

    const chatMessages = messages.map((message, index) =>
      <div key={index} className='msg'>
        <span className='user'>{message.user}</span>
        {': '}
        <span className='text'>{message.text}</span>
        <span className='status'>{`(${message.status})`}</span>
      </div>
    )

    return <div className='chat col-md-12'>
      <div className='input-group'>
        <span className='input-group-btn'>
          <button className='btn btn-default' type='button' onClick={this.handleSendClick}>Отправить</button>
        </span>
        <input type='text' ref='inputText' className='form-control' placeholder='Напишите сообщение...' />
      </div>
      <div className='messages'>{chatMessages}</div>
    </div>
  }
}
