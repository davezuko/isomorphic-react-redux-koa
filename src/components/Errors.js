import React, { PropTypes, Component } from 'react'
import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup'

export default class Errors extends Component {
  static propTypes = {
    errors: PropTypes.array
  }

  render() {
    const { errors } = this.props

    // отображаем последние 4 ошибки
    const errorList = errors.map((error, index) => {
      if (index > errors.length - 5) {
        return <div key={error.id} className='err'>{error.text}</div>
      }
    }
    )

    return <div className='errors'><CSSTransitionGroup
      transitionName='errors'
      transitionEnterTimeout={300}
      transitionLeaveTimeout={300}>
      {errorList}
    </CSSTransitionGroup>
    </div>
  }
}
