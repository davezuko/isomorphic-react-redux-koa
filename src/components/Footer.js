import React from 'react';

const Footer = () => (
  <footer
    className='container'>
    <p
      className='text-center'>
      React Redux Starter Kit Fullstack
    </p>
  </footer>
);

export default Footer;
