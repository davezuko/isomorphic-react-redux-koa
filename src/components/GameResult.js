import React, { PropTypes, Component } from 'react'
import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup'

export default class GameResult extends Component {
  static propTypes = {
    gameResult: PropTypes.string.isRequired
  }

  render() {
    const { gameResult } = this.props
    let result

    switch (gameResult) {
      case 'x':
        result = 'Победил игрок "X"'
        break
      case 'o':
        result = 'Победил игрок "O"'
        break
      case 'full':
        result = 'Ничья'
        break
      case 'disconnect':
        result = 'Соединение потеряно'
        break
      default:
        result = 'Не удалось определить победителя'
        break
    }

    return <CSSTransitionGroup
      transitionName='ttt'
      transitionAppear
      transitionAppearTimeout={500}
      transitionLeave={false}
      transitionEnterTimeout={500}>
      <div className='result'><h4>{result}</h4></div>
    </CSSTransitionGroup>
  }
}
