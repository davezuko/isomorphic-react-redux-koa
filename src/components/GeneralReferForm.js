import React from 'react'

import { Input } from 'react-bootstrap'

export default class GeneralReferForm extends React.Component {
  // static propTypes = {
  //   store: React.PropTypes.object.isRequired
  // }

  render () {
    return (
    <div>
      <form className='form-horizontal'>
        <Input type='text' label='Resume *' labelClassName='col-xs-3' wrapperClassName='col-xs-9' />
        <Input type='text' label='First Name' labelClassName='col-xs-3' wrapperClassName='col-xs-9' />
        <Input type='text' label='Last Name' labelClassName='col-xs-3' wrapperClassName='col-xs-9' />
        <Input type='select' label='Which derpartment are they most suited to? *'
          labelClassName='col-xs-3' wrapperClassName='col-xs-9'>
          <option value=''>All Departments</option>
          <option value='Eng'>Eng</option>
          <option value='Engineering, QA'>Engineering, QA</option>
          <option value='HR'>HR</option>
          <option value='Marketing'>Marketing</option>
          <option value='Sales'>Sales</option>
          <option value='Technology'>Technology</option>
        </Input>
        <Input type='textarea' label='Textarea' labelClassName='col-xs-3' wrapperClassName='col-xs-9' />
        <Input type='checkbox' label='Checkbox' wrapperClassName='col-xs-offset-2 col-xs-9' help='Offset is applied to wrapper.' />
      </form>
    </div>
    )
  }
}
