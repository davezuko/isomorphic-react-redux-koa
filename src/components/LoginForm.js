import React, { PropTypes } from 'react';

export default class LoginForm extends React.Component {
  static propTypes = {
    onFormSubmit: PropTypes.func.isRequired,
    isSubmitPending: PropTypes.bool.isRequired
  };

  constructor (props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      error: ''
    };

    this.onEmailInputChange = this.onEmailInputChange.bind(this);
    this.onPasswordInputChange = this.onPasswordInputChange.bind(this);
    this.onFormSubmit = this.onFormSubmit.bind(this);
  }

  onEmailInputChange (event) {
    this.setState({
      email: event.target.value
    });
  }

  onPasswordInputChange (event) {
    this.setState({
      password: event.target.value
    });
  }

  onFormSubmit (event) {
    event.preventDefault();

    const { email, password } = this.state;

    if (!email || !password)
      return this.setState({
        error: 'Please enter your email and password'
      });

    this.props.onFormSubmit({
      email,
      password
    });

    this.setState({
      email: '',
      password: '',
      error: ''
    });
  }

  get error () {
    return (
      <div className='form-group has-error'>
        <p className='help-block'>
          {this.state.error}
        </p>
      </div>
    );
  }

  get pending () {
    return (
      <div className='form-group'>
        <p>Please wait, logging in...</p>
      </div>
    );
  }

  render () {
    return (
       <form
        onSubmit={this.onFormSubmit}>

          <div className='form-group'>
            <label>Email</label>
            <input
              type="email"
              onChange={this.onEmailInputChange}
              value={this.state.email}
              className='form-control' />
          </div>

          <div className='form-group'>
            <label>Password</label>

            <input
              type="password"
              onChange={this.onPasswordInputChange}
              value={this.state.password}
              className="form-control" />
          </div>

          {this.state.error && this.error}

          <div className='form-group'>
            <input
              type='submit'
              value='Login'
              className='btn btn-inverse btn-lg btn-login' />
          </div>

          {this.props.isSubmitPending && this.pending}

       </form>
    );
  }
}

