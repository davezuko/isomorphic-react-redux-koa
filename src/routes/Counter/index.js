import { injectReducer } from '../../store/reducers'

export default (store) => ({
  path : 'counter',
  /*  Async getComponent is only invoked when route matches   */
  getComponent (nextState, cb) {
    /*  Webpack - use 'System.import' to create a split point
        and embed an async module loader (jsonp) when bundling   */
    Promise.all([
      System.import('./containers/CounterContainer'),
      System.import('./modules/counter')
    ]).then((modules) => {
      const Counter = modules[0].default
      const reducer = modules[1].default

      /*  Add the reducer to the store on key 'counter'  */
      injectReducer(store, { key: 'counter', reducer })

      /*  Return getComponent   */
      cb(null, Counter)
    })
  }
})
