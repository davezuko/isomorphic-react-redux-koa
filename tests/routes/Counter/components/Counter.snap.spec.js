import React from 'react'
import { Counter } from 'routes/Counter/components/Counter'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const component = <Counter counter={2} increment={jest.fn()} doubleAsync={jest.fn()} />
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})
