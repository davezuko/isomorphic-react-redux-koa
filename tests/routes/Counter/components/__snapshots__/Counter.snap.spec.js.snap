// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "0 auto",
    }
  }
>
  <h2>
    Counter: 
    2
  </h2>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Increment
  </button>
   
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Double (Async)
  </button>
</div>
`;
